PSICompile/ILCodeGen.cs

```diff
@@ -121,7 +178,7 @@ public class ILCodeGen : Visitor {
    public override void Visit (NUnary u) {
       u.Expr.Accept (this);
       string op = u.Op.Kind.ToString ().ToLower ();
-      op = op switch { "sub" => "neg", _ => op };
+      op = op switch { "sub" => "neg", "not" => "ldc.i4.0\n    ceq", _ => op };

```

Comment:
`WriteLn (not 2)` should print **-3** (which is the value you get when you flip all the bits in the binary representation of 2.  (OpCode = **not**)
`WriteLn (not false)` should print **True**.  (OpCode = **ldc.i4.0 \n ceq**)
Not should behave differently depending on whether the operand is boolean or integer.